iOS/Flows/Conversation/Cells/TimeMachineCollectionViewLayout.swift

```diff
@@ -353,55 +345,27 @@ class TimeMachineCollectionViewLayout: UICollectionViewLayout {
     }
 
-    func getCenterPoint(for section: SectionIndex,
-                        withYOffset yOffset: CGFloat,
-                        scale: CGFloat) -> CGPoint {
+    func getItemCenterPoint(in section: SectionIndex,
+                            withYOffset yOffset: CGFloat,
+                            scale: CGFloat) -> CGPoint {
         
         guard let collectionView = self.collectionView else { return .zero }
         
         let contentRect = CGRect(x: collectionView.contentOffset.x,
                                  y: collectionView.contentOffset.y,
                                  width: collectionView.bounds.size.width,
                                  height: collectionView.bounds.size.height)
-        
-        var centerPoint: CGPoint = .zero
-        
-        switch self.uiState {
-        case .read:
-            let additionalYOffset =  MessageContentView.bubbleHeight + Theme.ContentOffset.long.value
-            let centerY = contentRect.top + additionalYOffset
-            
-            centerPoint = CGPoint(x: contentRect.midX, y: centerY)
-            
-            if section == 0 {
-                centerPoint.y += self.itemHeight.half
-                centerPoint.y += yOffset
-                centerPoint.y += self.itemHeight.doubled * (1-scale)
-                centerPoint.y -= 100 - Theme.ContentOffset.short.value
-            } else {
-                centerPoint.y += 50
-                centerPoint.y += self.itemHeight.doubled - Theme.ContentOffset.short.value
-                centerPoint.y -= yOffset
-                centerPoint.y -= self.itemHeight.doubled * (1-scale)
-            }
-            
-        case .write:
-            let centerY = (contentRect.top)
-            centerPoint = CGPoint(x: contentRect.midX, y: centerY)
-            
-            if section == 0 {
-                centerPoint.y += self.itemHeight.half
-                centerPoint.y += yOffset
-                centerPoint.y += self.itemHeight.half * (1-scale)
-                centerPoint.y -= 21 - Theme.ContentOffset.short.value
-            } else {
-                if ScreenSize.current.rawValue <= ScreenSize.phoneMedium.rawValue {
-                    centerPoint.y -= 25
-                }
-                centerPoint.y += self.itemHeight.doubled - Theme.ContentOffset.short.value
-                centerPoint.y -= yOffset
-                centerPoint.y -= self.itemHeight.half * (1-scale)
-            }
+
+        var centerPoint = CGPoint(x: contentRect.midX, y: contentRect.top)
+
+        if section == 0 {
+            centerPoint.y += self.firstSectionTopY + self.itemHeight.half
+            centerPoint.y += self.itemHeight.half - self.itemHeight.half * scale
+            centerPoint.y += yOffset
+        } else {
+            centerPoint.y += self.secondSectionBottomY - self.itemHeight.half
+            centerPoint.y -= self.itemHeight.half - self.itemHeight.half * scale
+            centerPoint.y -= yOffset
```

Comment:
Now that you specify the section positions, the logic is a lot simpler and more generic.